src/install.rs

```diff
@@ -278,6 +287,373 @@ wiping them with `wipefs -a`.\n"
+        );
+    }
+    if config.insecure {
+        kargs.push_str(" coreos.inst.insecure");
+    }
+    if config.skip_reboot {
+        kargs.push_str(" coreos.inst.skip_reboot");
+    }
+    if let Some(url) = deferred_rootfs_url {
+        add_karg(&mut kargs, "coreos.live.rootfs_url", url.as_str());
+    };
+
+    eprintln!("Loading kernel and initramfs with arguments: {}", &kargs);
+    runcmd!(
+        "kexec",
+        "--load",
+        &kernel,
+        "--initrd",
+        fd_path(&final_initrd),
+        "--append",
+        &kargs
+    )?;
+
+    eprintln!("Pivoting");
+    runcmd!("systemctl", "kexec")?;
+
+    Ok(())
+}
+
+fn find_kernel_and_initrd() -> Result<(PathBuf, PathBuf)> {
+    const MODULES_DIR: &str = "/usr/lib/modules";
+    let mut kver_dir: Option<PathBuf> = None;
+    for entry in
+        read_dir(MODULES_DIR).with_context(|| format!("reading directory {}", MODULES_DIR))?
+    {
+        let entry = entry.with_context(|| format!("reading entry in directory {}", MODULES_DIR))?;
+        if let Some(prev) = kver_dir.replace(entry.path()) {
+            bail!(
+                "multiple directories found in {} (at least {} and {})",
+                MODULES_DIR,
+                prev.to_string_lossy(),
+                entry.path().to_string_lossy()
+            );
+        }
+    }
+    let kver_dir = kver_dir.context("no directories found in /usr/lib/modules")?;
+
+    // XXX: look for alternate names to be nice
+    Ok((kver_dir.join("vmlinuz"), kver_dir.join("initramfs.img")))
+}
+
+fn concat_initrds(bottom_initrd: &mut File, top_initrd: &mut File) -> Result<()> {
+    bottom_initrd
+        .seek(SeekFrom::End(0))
+        .context("seeking to end of bottom initramfs")?;
+    copy(top_initrd, bottom_initrd)?;
+    bottom_initrd
+        .seek(SeekFrom::Start(0))
+        .context("rewinding concatenated initramfs")?;
+    Ok(())
+}
+
+fn download_live_artifact(url: &Url, insecure: bool, retries: FetchRetries) -> Result<File> {
+    if insecure {
+        download_to_tempfile(url, retries)
+    } else {
+        let mut sig_url = url.clone();
+        sig_url.set_path(&format!("{}.sig", url.path()));
+        download_and_verify_to_tempfile(url, &sig_url, retries)
+    }
+}
+
+fn read_local_live_artifact(path: &str, insecure: bool) -> Result<File> {
+    let mut f = OpenOptions::new()
+        .read(true)
+        .open(path)
+        .with_context(|| format!("opening {}", path))?;
+    if insecure {
+        return Ok(f);
+    }
+
+    let sig = std::fs::read(format!("{}.sig", path))
+        .with_context(|| format!("opening signature file for {}", path))?;
+
+    let mut bf = BufReader::with_capacity(BUFFER_SIZE, &mut f);
+    let mut v = VerifyReader::new(&mut bf, Some(sig.as_slice()), VerifyKeys::Production)
+        .with_context(|| format!("creating verifier for {}", path))?;
+    copy(&mut v, &mut std::io::sink()).with_context(|| format!("reading {}", path))?;
+    v.verify_without_logging_failure()
```

Comment:
This method is intended for `try_` paths where we fall back to another code path on signature mismatch.  Since we do actually fail here, we should use the logging variant.